src/main/java/com/nvidia/spark/rapids/jni/OrcTimezoneInfo.java

```diff
@@ -0,0 +1,133 @@
+
+  // The following is Static fields and methods.
+  // The `orc_timezone_info.data` file is generated from `sun.util.calendar.ZoneInfo` on OpenJDK 8
+  // It first reads `transitions` and `offsets` fields from `ZoneInfo` via reflection.
+  // Then calculate the actual transition and offset values via:
+  // - actual transition = transition >> 12
+  // - actual offset = offsets[transition & 0x0FL]
+  // For more details, please refer to `sun.util.calendar.ZoneInfo` source code.
+
+  // There are 652 timezones, the file `orc_timezone_info.data` content:
+  // First part: 632 indices(integer) to store the start points in the file for each timezone.
+  // Seconds part: 632 serialized timezone info.
+  // Each timezone info content:
+  // rowOffset: 4 bytes
+  // num of transitions(num_transitions): 4 bytes
+  // transitions: num_transitions * 8 bytes
+  // num of offsets(num_offsets): 4 bytes
+  // offsets: num_offsets * 4 bytes
+  private static final String ORC_TIMEZONE_FILE = "orc_timezone_info.data";
+
+  // the mapped memory for the file
+  private static MappedByteBuffer serializedBuf = null;
+
+  static {
+    readTimezoneInfoFromFile();
+  }
+
+  private static void readTimezoneInfoFromFile() {
+    URL path = OrcTimezoneInfo.class.getClassLoader().getResource(ORC_TIMEZONE_FILE);
+    if (path == null) {
+      throw new RuntimeException("Can not find ORC timezone info file " + ORC_TIMEZONE_FILE);
+    }
+
+    try (RandomAccessFile file = new RandomAccessFile(path.getPath(), "r");
+         FileChannel fileChannel = file.getChannel()) {
+
+      if (fileChannel.size() > 2 * 1024 * 1024) { // > 2M
+        throw new RuntimeException("Failed to load ORC timezone info, file is too large > 2M.");
+      }
+
+      // Map the file into memory
+      serializedBuf = fileChannel.map(FileChannel.MapMode.READ_ONLY, 0, fileChannel.size());
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to load ORC timezone info file " + ORC_TIMEZONE_FILE, e);
+    }
+  }
+
+  /**
+   * Get timezone info for the specified timezone Id
+   * @param timezoneId timezone Id
+   * @return timezone info
+   */
+  public static OrcTimezoneInfo get(String timezoneId) {
+    int index = Arrays.binarySearch(timezoneIds, timezoneId);
+    if (index < 0) {
+      throw new IllegalArgumentException("Timezone ID not found: " + timezoneId);
+    }
+
+    // shallow copy
+    ByteBuffer buf = serializedBuf.duplicate();
+    buf.order(ByteOrder.BIG_ENDIAN);
+
+    int timezoneInfoOffsetInFile = buf.getInt(Integer.BYTES * index);
+    buf.position(timezoneInfoOffsetInFile);
+
+    int rawOffsets = buf.getInt();
+
+    int numTransitions = buf.getInt();
+    long[] transitions = new long[numTransitions];
+    for (int i = 0; i < numTransitions; ++i) {
+      transitions[i] = buf.getLong();
+    }
+
+    int numOffsets = buf.getInt();
+    int[] offsets = new int[numOffsets];
+    for (int i = 0; i < numOffsets; ++i) {
+      offsets[i] = buf.getInt();
+    }
```

Comment:
Ok, I think I understand how this has been arranged.  I see how this should work.